xgboost_ray/matrix.py

```diff
@@ -48,6 +48,25 @@ def concat_dataframes(dfs: List[Optional[pd.DataFrame]]):
     return pd.concat(filtered, ignore_index=True, copy=False)
 
 
+def ensure_sorted_by_qid(df: pd.DataFrame, qid: Data) -> pd.DataFrame:
+    _qid = None
+    if isinstance(qid, str):
+        _qid = df[qid]
+    elif isinstance(qid, np.ndarray):
+        _qid = pd.Series(qid)
+    elif isinstance(qid, pd.DataFrame):
+        _qid = qid.iloc[:, 0]
```

Comment:
can we raise an exception if it has more than 1 column?

Reply:
sure